test/unit/specs/components/v-content-link.spec.js

```diff
@@ -0,0 +1,48 @@
+import VContentLink from '~/components/VContentLink/VContentLink.vue'
+import { render, screen } from '@testing-library/vue'
+import userEvent from '@testing-library/user-event'
+import VueI18n from 'vue-i18n'
+
+const enMessages = require('~/locales/en.json')
+const i18n = new VueI18n({
+  locale: 'en',
+  fallbackLocale: 'en',
+  messages: { en: enMessages },
+})
+
+describe('VContentLink', () => {
+  let options = {}
+
+  beforeEach(() => {
+    options = {
+      props: { mediaType: 'image', resultsCount: 123 },
+      mocks: {
+        $nuxt: {
+          context: {
+            i18n,
+          },
+        },
+      },
+    }
+  })
+
+  it('is not selected by default', () => {
+    render(VContentLink, options)
+    const btn = screen.getByRole('radio')
+    expect(btn).not.toHaveAttribute('aria-checked')
+  })
+
+  it('is marked as selected when indicated with the isSelected prop', () => {
+    options.props.isSelected = true
+    render(VContentLink, options)
+    const btn = screen.getByRole('radio')
+    expect(btn).toHaveAttribute('aria-checked')
+  })
+
+  it('should emit selected event on click', () => {
+    const { emitted } = render(VContentLink, options)
+    const btn = screen.getByText('search-tab.image')
+    userEvent.click(btn)
+    expect(emitted().selected).toBeTruthy()
+  })
```

Comment:
Non-blocker, kind of philosophical question 😅 I kind of wonder if instead of emitting our own custom event when the button is clicked, if it wouldn't be easier to just `v-on="$attrs"` on the `button` in the component so that consumers can just attach whatever events they want to it instead of just this `selected` event (which users will have to remember what the signature of it is every time they want to use this component).

Reply:
I was trying before and the native event wasn't working for me, then I realized it was using the wrong instance property and I had to use `v-on="$listeners"`, thanks for making me look again at this! It should be a transparent component for events now 😄